tools/src/versioning/android/versionCxx/index.ts

```diff
@@ -0,0 +1,140 @@
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs-extra';
+import glob from 'glob-promise';
+import path from 'path';
+
+import { ANDROID_DIR, PACKAGES_DIR, EXPOTOOLS_DIR } from '../../../Constants';
+import { getListOfPackagesAsync, Package } from '../../../Packages';
+import { transformFileAsync, transformString } from '../../../Transforms';
+
+const CXX_EXPO_MODULE_PATCHES_DIR = path.join(
+  EXPOTOOLS_DIR,
+  'src',
+  'versioning',
+  'android',
+  'versionCxx',
+  'patches'
+);
+
+/**
+ * Executes the versioning for expo-modules with cxx code.
+ *
+ * Currently, it is a patch based process.
+ * we patch build files directly in `packages/{packageName}`,
+ * build the share libraries in place and copy back to versioned jniLibs folder.
+ * To add an module for versioning,
+ * please adds a corresponding `tools/src/versioning/android/versionCxx/patches/{packageName}.patch` patch file.
+ */
+export async function versionCxxExpoModulesAsync(version: string) {
+  const packages = await getListOfPackagesAsync();
+  const versionablePackages = packages.filter((pkg) => isVersionableCxxExpoModule(pkg));
+
+  await Promise.all(
+    versionablePackages.map(async (pkg) => {
+      const { packageName } = pkg;
+      const abiName = `abi${version.replace(/\./g, '_')}`;
+      const versionedAbiRoot = path.join(ANDROID_DIR, 'versioned-abis', `expoview-${abiName}`);
+
+      const patchContent = await getTransformPatchContentAsync(packageName, abiName);
+      const patchStripingArgs = '-p3'; // -pN passing to the `patch` command for striping slashed prefixes
+
+      // Applies versioning patches
+      let procPromise = spawnAsync('patch', [patchStripingArgs], {
+        cwd: path.join(PACKAGES_DIR, packageName),
+      });
+      procPromise.child.stdin?.write(patchContent);
+      procPromise.child.stdin?.end();
+      await procPromise;
```

Comment:
What about extracting to a function? 🤔
 ```suggestion
      await applyPatchForPackage(packageName, patchContent);
```